wikimedia/wikipedia-ios
# [HOLD] Native Source Editor - Bold/italic button selection, insertion & deletion support

**Phabricator:** 
https://phabricator.wikimedia.org/T347925 (button selection states for bold and italic, formatter inserts / deletions for bold & italic)

### Notes
_Note: This PR is dependent on #4679. Marking as hold while we wait on that._

This PR automatically sets the bold and italic button in selected or unselected states, depending on where the text view selection or cursor is. It also adds the bold / italic wikitext insertion and deletion logic upon button tap. 

### Test Steps
1. Go to editor on Staging.
2. Scroll down and select some already-formatted bold text. Confirm bold buttons in keyboard menus show as selected.
3. Tap bold button to unselect. Confirm bold formatting is automatically removed around selection.
4. Tap bold button again to select. Confirm bold formatting is automatically added around selection.
5. Place cursor somewhere not within a bold range. Tap the bold button. Confirm you see bold wikitext inserted around cursor.
6. Repeat steps 1 - 5 for italics.
